src/sentry/tasks/auth.py

```diff
@@ -100,36 +104,46 @@ def remove_member(member):
                 logger.info(
                     f"{self.log_label} noncompliant user removed from org", extra=logging_data
                 )
-                AuditLogEntry.objects.create(
-                    actor=actor,
-                    actor_key=actor_key,
+                create_audit_entry_from_user(
+                    user=actor,
+                    actor_key_id=actor_key_id,
                     ip_address=ip_address,
                     event=audit_log.get_event_id("MEMBER_PENDING"),
-                    data=member.get_audit_log_data(),
+                    data=org_member.get_audit_log_data(),
                     organization_id=org_id,
                     target_object=org_id,
                     target_user_id=user.id,
                 )
+                # Refresh the org member to ensure we always properly generate an invite link
+                org_member.refresh_from_db()
```

Comment:
So I removed an n+1 rpc, and replaced it with an n+1 org member refresh 😅  I didn't want to change the invite link behavior to always return a link as it is used in a number of places so I kept this mostly intact.

I could duplicate this method on the rpc org member though if we really want to reduce the number of queries here.

Reply:
Database refreshing is *much* faster than an RPC across the ocean. I don't think 2fa requirements change with enough frequency that we need to sweat the additional DB queries.